daemon/cluster/utils.go

```diff
@@ -61,3 +65,40 @@ func removingManagerCausesLossOfQuorum(reachable, unreachable int) bool {
 func isLastManager(reachable, unreachable int) bool {
 	return reachable == 1 && unreachable == 0
 }
+
+// newErrWithHTTPCodeFromGRPC converts an error from gRPC into an error with HTTP status code
+func newErrWithHTTPCodeFromGRPC(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	switch grpc.Code(err) {
+	case codes.OK: // code 0
+		return nil
+	case codes.InvalidArgument: // code 3
+		return errors.NewBadRequestError(err)
+	case codes.DeadlineExceeded: // code 4
+		return errors.NewErrorWithStatusCode(err, http.StatusRequestTimeout)
+	case codes.NotFound: // code 5
+		return errors.NewRequestNotFoundError(err)
+	case codes.AlreadyExists: // code 6
+		return errors.NewRequestConflictError(err)
```

Comment:
Conflict is a specific http error. This should be a 400 or some other error. There is already an open issue about the incorrect usage of 409. @justincormack